tests/editorhelper_test.cpp

```diff
@@ -38,477 +94,233 @@ static bool writeFile(const char* filename, void*& result, int length) {
-  EXPECT_TRUE(out_img.pixelFormat == in_img.pixelFormat);
-#ifdef DUMP_OUTPUT
-  if (!writeFile("rotated_180.y", out_img.data, outSize)) {
-    std::cerr << "unable to write output file" << std::endl;
-  }
-#endif
+TEST_P(EditorHelperTest, Rotate) {
+  initImageHandle(&img_a, width, height, fmt);
+  ASSERT_TRUE(loadFile(filename.c_str(), &img_a)) << "unable to load file " << filename;
+  auto dst = apply_rotate(&img_a, 90);
+  dst = apply_rotate(dst.get(), 90);
+  dst = apply_rotate(dst.get(), 180);
+  dst = apply_rotate(dst.get(), 270);
+  dst = apply_rotate(dst.get(), 90);
+  dst = apply_rotate(dst.get(), 90);
+  dst = apply_rotate(dst.get(), 270);
+  ASSERT_NO_FATAL_FAILURE(compareImg(&img_a, dst.get()));
 }
 
-TEST_F(EditorHelperTest, rotatingGreyImage270) {
-  jpegr_uncompressed_struct in_img;
-  jpegr_uncompressed_struct out_img;
-
-  in_img.data = mGreyImage.buffer.get();
-  in_img.width = IMAGE_WIDTH;
-  in_img.height = IMAGE_HEIGHT;
-  in_img.pixelFormat = ultrahdr_pixel_format::ULTRAHDR_PIX_FMT_MONOCHROME;
-
-  std::unique_ptr<uint8_t[]> out_img_data;
-  int outSize = IMAGE_WIDTH * IMAGE_HEIGHT;
-  out_img_data.reset(new uint8_t[outSize]);
-  out_img.data = out_img_data.get();
-  EXPECT_TRUE(rotate(&in_img, 270, &out_img)== JPEGR_NO_ERROR);
-  EXPECT_TRUE(out_img.width = IMAGE_HEIGHT);
-  EXPECT_TRUE(out_img.height = IMAGE_WIDTH);
-  EXPECT_TRUE(out_img.pixelFormat == in_img.pixelFormat);
-#ifdef DUMP_OUTPUT
-  if (!writeFile("rotated_270.y", out_img.data, outSize)) {
-    std::cerr << "unable to write output file" << std::endl;
-  }
-#endif
+TEST_P(EditorHelperTest, Mirror) {
+  initImageHandle(&img_a, width, height, fmt);
+  ASSERT_TRUE(loadFile(filename.c_str(), &img_a)) << "unable to load file " << filename;
+  auto dst = apply_mirror(&img_a, UHDR_MIRROR_VERTICAL);
+  dst = apply_mirror(dst.get(), UHDR_MIRROR_VERTICAL);
+  dst = apply_mirror(dst.get(), UHDR_MIRROR_HORIZONTAL);
+  dst = apply_mirror(dst.get(), UHDR_MIRROR_HORIZONTAL);
+  ASSERT_NO_FATAL_FAILURE(compareImg(&img_a, dst.get()));
 }
 
-TEST_F(EditorHelperTest, resizeYuvImageUp) {
-  jpegr_uncompressed_struct in_img;
-  jpegr_uncompressed_struct out_img;
-
-  in_img.data = mYuvImage.buffer.get();
-  in_img.width = IMAGE_WIDTH;
-  in_img.height = IMAGE_HEIGHT;
-  in_img.colorGamut = ultrahdr_color_gamut::ULTRAHDR_COLORGAMUT_BT709;
-  in_img.pixelFormat = ultrahdr_pixel_format::ULTRAHDR_PIX_FMT_YUV420;
-
-  std::unique_ptr<uint8_t[]> out_img_data;
-  int out_width = IMAGE_WIDTH * 3 / 2;
-  int out_height = IMAGE_HEIGHT * 3 / 2;
-  int outSize = out_width * out_height * 3 / 2;
-  out_img_data.reset(new uint8_t[outSize]);
-  out_img.data = out_img_data.get();
-  EXPECT_TRUE(resize(&in_img, out_width, out_height, &out_img)== JPEGR_NO_ERROR);
-  EXPECT_TRUE(out_img.width = out_width);
-  EXPECT_TRUE(out_img.height = out_height);
-  EXPECT_TRUE(out_img.colorGamut == in_img.colorGamut);
-  EXPECT_TRUE(out_img.pixelFormat == in_img.pixelFormat);
+TEST_P(EditorHelperTest, Crop) {
```

Comment:
Better to have one more test for applying multiple effects in a row, and verify the process doesn't crash, result is correct

Reply:
Done